pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -188,24 +188,28 @@ func (c *Controller) extractGatewaysInner(svc *model.Service) bool {
 		}
 	}
 
-	gwsChanged := len(c.networkGateways[svc.Hostname][network]) != len(gws)
-	if !gwsChanged {
```

Comment:
@hzxuzhonghu previously we only checked if every element in the new set was present in the old set, this change makes us check both ways

it's unlikely to matter, usually we'd see a `len` change instead of things being swapped in place. 

Also, extra test coverage is always nice.

Reply:
IC, lgtm except the function detail